resdk/analysis/prepare_geo.py

```diff
@@ -0,0 +1,170 @@
+"""Prepare GEO."""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import time
+
+from resdk.resources.utils import (
+    get_resource_collection, get_samples, get_resolwe, is_background
+)
+
+__all__ = ('prepare_geo_chipseq', 'prepare_geo_rnaseq', 'prepare_geo')
+
+
+def get_name_collection(collection_ids, res):
+    """Get the ``name`` input for ``prepare_geo`` processes.
+
+    This function returns collection name and collection id if there
+    is only one element in 'collection_ids' (all the samples are in the
+    same collection). Otherwise it returns current date as ``name``.
+
+    :param set collection_ids: set of collection id's of samples
+    :param res: Resolwe instance
+    :type res: :class:`resdk.resolwe.Resolwe`
+
+    """
+
+    collection_ids = list(collection_ids)
+
+    if len(collection_ids) == 1 and collection_ids[0]:
+        collection = res.collection.get(collection_ids[0])
+        name = collection.name
+    else:
+        collection = None
+        name = time.strftime("%d-%m-%Y")
+
+    return name, collection
+
+
+def prepare_geo_chipseq(resource, name=None):
+    """Run ``Prepare GEO - ChIP-Seq`` process on the resource.
+
+    This method can be used to run ``Prepare GEO - ChIP-Seq`` process
+    on a single collection or a list of samples.
+
+    :param resource: resource on which prepare_geo_chipseq will be run
+    :param str name: name of the prepare GEO tarball and table
+
+    """
+
+    reads = []
+    macs14 = []
+    relations = []
+
+    samples = get_samples(resource)
+    resolwe = get_resolwe(*samples)
+    collection_ids = set()
+
+    for sample in samples:
+        reads.append(sample.get_reads().id)
+
+        if is_background(sample):
+            continue
+
+        macs_list = sample.get_macs()
+        if not macs_list:
+            raise ValueError(
+                "Sample {} has no `macs14` data object!"
```

Comment:
Join the two lines as they don't exceed 99 characters together. Fix this in all exceptions.